### DISCUSSION TITLE
关于Memory warning 256.00 M的警告

### DISCUSSION THREAD

**Discussion by ericzhanchina** [cloudwu/skynet#1792] [General]
产品上线了，目前用户不多，我用了两台4CPU  16G的服务器做Game1  Game2服，跑了两天没问题，每个服务器在线一般3000人，结果晚上服务器开始出现了无法访问的问题，我看日志有Memory warning 256.00 M的警告。

```
08/09/23 21:51:09.88 [:000005db] KILL self
08/09/23 21:51:10.01 [:000005c1] KILL self
08/09/23 21:51:10.52 [:0000000f] Memory warning 256.00 M
08/09/23 21:51:11.55 [:00000d41] KILL self
08/09/23 21:51:12.19 [:0000102c] KILL self
08/09/23 21:51:13.56 [:00000e99] KILL self
08/09/23 21:51:14.36 [:00000f82] KILL self
```

这个0000000f就是gated服，现在每个msgagent都会请求gated，其中有两个是挂起的task，一个请求当前地图的野怪的刷新情况，一个是请求是否有推送消息文本，这里我都用的skynet.response处理的，只有有野怪刷新，或者有推送消息，才会回给gated，不知道是不是这个原因造成gated太大的，是不是达到一定程度就会无法响应服务请求了。我这样设计有问题吗？请有经验的大佬指点一下，这样不行的话，要改回轮询的方式吗？

**Comment by cloudwu** [comment]
感觉无法响应（cpu 问题）和该警告关系不大。这里时间戳也是连续的。还需定位其它问题。
如果怀疑出在 gated 服务，可进一步查询 gated 所占 cpu 。

**Reply by ericzhanchina** [reply]
云风，这种警告是说这个服务占的内存太大了是吗？我现在这种response挂起的方式您觉得合适吗，还是说改成客户端不断来轮询服务器？

**Reply by ericzhanchina** [reply]
还有一个问题，我现在登录到控制台，因为后台几千个服务，想查看gated的情况也看不了，用Info命令也不行，这个我怎么才能看到gated的内存占用，task数目等等呢，原来服务少的时候，拉上去就可以看到了。请教一下。

**Comment by hanxi** [comment]
> 还有一个问题，我现在登录到控制台，因为后台几千个服务，想查看gated的情况也看不了，用Info命令也不行，这个我怎么才能看到gated的内存占用，task数目等等呢，原来服务少的时候，拉上去就可以看到了。请教一下。

可以用curl发指令给控制台，然后用管道传给less命令

**Reply by hanxi** [reply]
这里有curl的例子 https://github.com/cloudwu/skynet/wiki/DebugConsole

**Reply by ericzhanchina** [reply]
感谢大佬得指点

**Comment by cloudwu** [comment]
内存警告是方便你 oom 崩溃后，或用到交换分区时，事后有 log 备查。
